components/file_preview_modal/file_preview_modal_main_actions/file_preview_modal_main_actions.tsx

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {memo, useEffect, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {Tooltip} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import OverlayTrigger from '../../overlay_trigger';
+import Constants from '../../../utils/constants';
+
+import './file_preview_modal_main_actions.scss';
+
+import {GlobalState} from '../../../types/store';
+import {getFilePublicLink} from 'mattermost-redux/actions/files';
+import {getFilePublicLink as selectFilePublicLink} from 'mattermost-redux/selectors/entities/files';
+import {copyToClipboard} from '../../../utils/utils';
+import {FileInfo} from 'mattermost-redux/types/files';
+
+interface DownloadLinkProps {
+    download?: string;
+}
+
+interface Props {
+    usedInside?: 'Header' | 'Footer';
+    showOnlyClose?: boolean;
+    showClose?: boolean;
+    filename: string;
+    fileURL: string;
+    fileInfo: FileInfo;
+    enablePublicLink: boolean;
+    canDownloadFiles: boolean;
+    onGetPublicLink?: () => void;
```

Comment:
do we use this prop?

Reply:
Yeah except onGetPublicLink?: () => void will remove this one. Thanks